pkg/dependencies/customresource.go

```diff
@@ -14,13 +14,13 @@ const (
 )
 
 var (
-	CustomResourceClientInitializer = dependency[*apiextensionsclientset.Clientset]{
+	CustomResourceClientInitializer = dependency[apiextensionsclientset.Interface]{
 		dep:     NewCRDClient,
 		applier: checks.WithCustomResourceClient,
 	}
 )
 
-func NewCRDClient(ctx context.Context, http bool, token string, kubeconfig string) (*apiextensionsclientset.Clientset, *checks.Check) {
+func NewCRDClient(ctx context.Context, endpoint string, insecure bool, http bool, token string, kubeconfig string) (apiextensionsclientset.Interface, *checks.Check) {
```

Comment:
nit: with this many repeated params, might be worth having a struct instead

Reply:
yep, going to refactor in the next PR.